src/redux/slices/workspace-helpers.ts

```diff
@@ -27,6 +29,28 @@ export const isDiagramPanel = (panelType: PanelType): boolean => {
     );
 };
 
+// Find parent NAD panel for a given SLD panel
+export const findNadForSld = (workspace: Workspace, sldPanelId: UUID): UUID | null => {
+    const nadPanel = Object.values(workspace.panels).find(
+        (panel) =>
+            panel.type === PanelType.NAD &&
+            (panel.metadata as NADPanelMetadata | undefined)?.associatedVoltageLevelPanels?.includes(sldPanelId)
+    );
+    return nadPanel?.id ?? null;
+};
```

Comment:
how it this different from selectNadForSld in src/redux/slices/workspace-selectors.ts